app/views/transaction.py

```diff
@@ -0,0 +1,766 @@
+                                             message=errors,
+                                             success=False,
+                                             status=HTTPStatus.BAD_REQUEST)
+                return response.error_response()
+
+            # fund transfer type - funds transfer
+            fund_transfer_data = FundTransfer(
+                from_account=data['from_account'],
+                to_account=data['to_account'])
+
+            db.session.add(fund_transfer_data)
+            db.session.commit()
+            result = fund_transfer_schema.dump(fund_transfer_data)
+
+            from_bank_account = BankAccount.query.filter(BankAccount.account_number == data['from_account']).first()
+            if not from_bank_account:
+                raise BankAccountObjectNotFound("Bank account details does not exist")
+
+            to_bank_account = BankAccount.query.filter(BankAccount.account_number == data['to_account']).first()
+            if not to_bank_account:
+                raise BankAccountObjectNotFound("Bank account details does not exist")
+
+            transaction_type = TransactionType.query.filter(TransactionType.transaction_type == "debit").first()
+            if not transaction_type:
+                raise TransactionTypeObjectNotFound("Transaction type does not exist")
+
+            if from_bank_account.account_balance - 1000 - data['transaction_amount'] > 0:
+                from_bank_account.account_balance -= data['transaction_amount']
+                db.session.add(from_bank_account)
+                to_bank_account.account_balance += data['transaction_amount']
+                db.session.add(to_bank_account)
+                transaction_status = "success"
+            else:
+                transaction_status = "failed"
+
+            account_transaction_data = AccountTransactionDetails(
+                transaction_amount=data['transaction_amount'],
+                transaction_status=transaction_status,
+                bank_account_id=from_bank_account.id,
+                transaction_type_id=transaction_type.id,
+                fund_transfer_id=result.get('id'),
+                fund_transfer_info="Funds Transfer"
+            )
+
+            db.session.add(account_transaction_data)
+            db.session.commit()
+            account_transaction_details_schema.dump(account_transaction_data)
+
+            logger.info("Response for post request for fund transfer {}".format(result))
+            if transaction_status == 'success':
+                response = ResponseGenerator(data=result,
+                                             message="Fund transfer successfully",
+                                             success=True,
+                                             status=HTTPStatus.OK)
+            else:
+                response = ResponseGenerator(data=result,
+                                             message="Fund transfer declined",
```

Comment:
If we are saying fund transfer was declined then we should give reason why it is declined.

Reply:
Made the changes as suggested.